src/Middleware/Localization/test/UnitTests/RequestLocalizationMiddlewareTest.cs

```diff
@@ -0,0 +1,70 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Localization;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.Extensions.Localization;
+
+public class RequestLocalizationMiddlewareTest
+{
+    [Theory]
+    [InlineData("zh-Hans-CN")]
+    [InlineData("zh-Hans")]
+    [InlineData("zh-CN")]
+    [InlineData("zh-Hant-TW")]
+    [InlineData("zh-Hant")]
+    [InlineData("zh-TW")]
+    public async Task RequestLocalizationMiddleware_ShouldFallBackToParentCultures_RegradlessOfHyphenSeparatorCheck(string requestedCulture)
+    {
+        using var host = new HostBuilder()
+            .ConfigureWebHost(webHostBuilder =>
+            {
+                webHostBuilder
+                .UseTestServer()
+                .Configure(app =>
+                {
+                    var supportedCultures = new[] { "ar", "en", "zh" };
+
+                    app.UseRequestLocalization(options =>
+                    {
+                        options.AddSupportedCultures(supportedCultures)
+                            .AddSupportedUICultures(supportedCultures)
+                            .AddInitialRequestCultureProvider(new CookieRequestCultureProvider
+                            {
+                                CookieName = "Preferences"
+                            });
+                    });
+
+                    app.Run(async context =>
+                    {
+                        var requestCulture = context.Features.Get<IRequestCultureFeature>();
+
+                        Assert.Equal("zh", requestCulture.RequestCulture.Culture.Name);
```

Comment:
This assertion doesn't really prove that the code is walking the parent hierarchy vs. chopping up the string. Even though you have the ```zh-TW``` test case in here, what you really want to assert is that it can find ```zh-Hant``` as a supported culture (implies that it is in the list of supported cultures).

Reply:
> what you really want to assert is that it can find zh-Hant as a supported culture (implies that it is in the list of supported cultures).

This should be easy if the `zh-Hant` is supported culture